test/functional/page_objects/dashboard_page_controls.ts

```diff
@@ -414,13 +422,42 @@ export class DashboardPageControls extends FtrService {
         return await option.getVisibleText();
       })
     );
+    return invalidSelections;
+  }
+
+  public async optionsListPopoverGetAvailableOptions() {
```

Comment:
I had to modify the logic for this function so that it would work when `showOnlySelected` was `true` - I could no longer rely on the cardinality (since the cardinality != the number of selections) and there is no way to see the number of selections from within the popover component (and I cannot access the options list button component without an ID). 

SO, long story short, just had to modify the logic a bit to detect when we wrap around to the first element again - this will tell us that we've reached the end of our list :shrug: